eng/scripts/TypeSpec-Validation.ps1

```diff
@@ -4,51 +4,106 @@ param (
+$typespecFolders, $checkedAll = &"$PSScriptRoot/Get-TypeSpec-Folders.ps1" `
+  -BaseCommitish:$BaseCommitish `
+  -TargetCommitish:$TargetCommitish `
+  -CheckAll:$CheckAll
+
+# Use -FolderCount to do performance testing on Typespec-Validation.ps1 with 
+# -CheckAll set to $true. This constrains the typeSpec folders to the first N 
+# folders. 
+if ($FolderCount) { 
+  $typespecFolders = $typespecFolders[0..($FolderCount - 1)]
+}
 
 $typespecFoldersWithFailures = @()
 if ($typespecFolders) {
-  $typespecFolders = $typespecFolders.Split('',[System.StringSplitOptions]::RemoveEmptyEntries)
-  foreach ($typespecFolder in $typespecFolders) {
-    LogGroupStart "Validating $typespecFolder"
+  # Group typespec folders by "service" (the first folder in the path). This
+  # parallel validation at the level of the service.
+  # Example: 'specification/ai/DocumentIntelligence' -> 'specification/ai'.
+  $serviceFolders = $typespecFolders | Group-Object -Property { Split-Path $_ -Parent }
+  
+  Write-Host "Starting per-service parallel validation..."
+
+  $serviceFolders `
+  | ForEach-Object -ThrottleLimit $Parallelism -Parallel {
+    $service = $_.Name
+    $typespecFolders = $_.Group
 
-    if ($checkedAll) {
-      $suppression = Get-Suppression "TypeSpecValidationAll" $typespecFolder
-      if ($suppression) {
-        $reason = $suppression["reason"] ?? "<no reason specified>"
-        LogInfo "Suppressed: $reason"
-        LogGroupEnd
+    $result = @{
+      Service = $service;
+      Suppressed = @();
+      Logs = @{};
+      Errors = @{};
+    }
+  
+    foreach ($typespecFolder in $typespecFolders) {  
+      if ($checkedAll) {
+        $suppression = Get-Suppression "TypeSpecValidationAll" $typespecFolder
+        if ($suppression) {
+          $reason = $suppression["reason"] ?? "<no reason specified>"
+          $result["Suppressed"] += "Suppressed: $typespecFolder $reason"
+          continue
+        }
+      }
+  
+      $result.Logs[$typespecFolder] = @("npm exec --no -- tsv $typespecFolder")
+  
+      if ($DryRun) {
         continue
       }
+  
+      $result.Logs[$typespecFolder] += npm exec --no -- tsv $typespecFolder 2>&1
+      if ($LASTEXITCODE) {
+        $errorString = "TypeSpec Validation failed for project $typespecFolder run the following command locally to validate."
+        $errorString += "`n > npm ci"
+        $errorString += "`n > npx tsv $typespecFolder"
+        $errorString += "`nFor more detailed docs see https://aka.ms/azsdk/specs/typespec-validation"
+  
+        $result.Errors[$typespecFolder] = $errorString
+      }
+  
+      if ($GitClean) {
+        # TODO: Ensure this is proper and correct. Can something in service A
+        # affect files in Service B? Can they impact files in the root? Worth 
+        # examining.
+        git restore $service | Out-Null
+        git clean -df $service | Out-Null
```

Comment:
Still checking on this.

Reply:
A test run of the original script reveals no changes OUTSIDE of the `specification/service` folder when running tests. Cleaning at the `specification/service` level between `typeSpecFolder`s is expected to work .